app/src/main/java/com/infomaniak/mail/ui/alertDialogs/BaseAlertDialog.kt

```diff
@@ -41,7 +41,7 @@ abstract class BaseAlertDialog(@ActivityContext private val activityContext: Con
     fun startLoading() {
         alertDialog.setCancelable(false)
         negativeButton.isEnabled = false
-        Utils.createRefreshTimer(onTimerFinish = positiveButton::showProgress).start()
+        positiveButton.showProgressAfterTimer()
```

Comment:
Ok, actually, after reading a bit, this comment applies to all other uses of `showProgressAfterTimer()`. Was there a reason to reinstanciate the timer everytime or not really?

Reply:
Well; creating a lazy instance of the timer seems to create memory leak because the timer keeps a reference ot the button in their callback

Reply:
Also we need it to be called when the alert is shown, to have the reference of the button because we can't have it otherwise
With a bit of logic change I think it could be possible, but is the Timer object worth complexifying the code to have a single instance ? I don't really know to which extend it is costly to create